sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -111,6 +111,17 @@ object RapidsPluginUtils extends Logging {
     }
   }
 
+  def detectMultiplePluginJars(): Unit = {
+    val classloader = classOf[com.nvidia.spark.rapids.SparkShimServiceProvider].getClassLoader
+    val rapidsJarURLs = classloader.getResources("rapids4spark-version-info.properties")
+        .asScala.toList
+    if (rapidsJarURLs.size > 1) {
+      val rapidsJars = rapidsJarURLs.map(_.toString.split("!").head).mkString(",")
+      throw new RuntimeException(s"Multiple rapids4spark jars found in the classpath: " +
+          s"$rapidsJars. Please make sure there is only one rapids4spark jar in the classpath.")
+    }
```

Comment:
The [spark-shell experiment](https://github.com/NVIDIA/spark-rapids/issues/7309#issuecomment-1790930071) did not accommodate for the fact that we might get a duplication spread over user and system classloaders. So one in a system classloader via extraClassPath or because it's in `$SPARK_HOME/jars`, and another one via `spark.jars`.
This would not be caught:
```
~/dist/spark-3.3.0-bin-hadoop3-scala2.13/bin/spark-shell \
  --jars /tmp/rapids-4-spark_2.13-23.12.0-SNAPSHOT-cuda11.jar \
  --driver-class-path scala2.13/dist/target/rapids-4-spark_2.13-23.12.0-SNAPSHOT-cuda11.jar \
  --conf spark.plugins=com.nvidia.spark.SQLPlugin 
```


nit: We have a constant PLUGIN_PROPS_FILENAME for "rapids4spark-version-info.properties"

nit: conditional throwing RuntimeException can be replaced with require

Reply:
Thanks for review, I fixed the nits.

Reply:
It should be covered after getShimClassLoader change, when I copy the freshly-built jar to /tmp and execute

```
~/dist/spark-3.3.0-bin-hadoop3-scala2.13/bin/spark-shell --jars /tmp/rapids-4-spark_2.13-23.12.0-SNAPSHOT-cuda11.jar --driver-class-path scala2.13/dist/target/rapids-4-spark_2.13-23.12.0-SNAPSHOT-cuda11.jar --conf spark.plugins=com.nvidia.spark.SQLPlugin
``` 

I see the failure as expected

Reply:
Yes, you are right, I can repro now. I guess spark-shell loaded an old jar so it didn't fail when I tested it.